modules/candidate_parameters/js/candidateInfo.js

```diff
@@ -0,0 +1,308 @@
+      caveatOptions: {
+        true: "True",
+        false: "False"
+      },
+      Data: [],
+      formData: {},
+      updateResult: null,
+      errorMessage: null,
+      isLoaded: false,
+      loadedData: 0
+    };
+  },
+  componentDidMount: function () {
+    var that = this;
+    $.ajax(this.props.dataURL, {
+      dataType: 'json',
+      xhr: function () {
+        var xhr = new window.XMLHttpRequest();
+        xhr.addEventListener("progress", function (evt) {
+          that.setState({
+            loadedData: evt.loaded
+          });
+        });
+        return xhr;
+      },
+      success: function (data) {
+        that.setState({
+          Data: data,
+          isLoaded: true
+        });
+      },
+      error: function (data, errorCode, errorMsg) {
+        that.setState({
+          error: 'An error occurred when loading the form!'
+        });
+      }
+    });
+  },
+  setFormData: function (formElement, value) {
+    var formData = this.state.formData;
+    formData[formElement] = value;
+    this.setState({
+      formData: formData
+    });
+  },
+  onSubmit: function (e) {
+    e.preventDefault();
+  },
+  render: function () {
+    if (!this.state.isLoaded) {
+      if (this.state.error !== undefined) {
+        return React.createElement(
+          "div",
+          { className: "alert alert-danger text-center" },
+          React.createElement(
+            "strong",
+            null,
+            this.state.error
+          )
+        );
+      }
+
+      return React.createElement(
+        "button",
+        { className: "btn-info has-spinner" },
+        "Loading",
+        React.createElement("span", { className: "glyphicon glyphicon-refresh glyphicon-refresh-animate" })
+      );
+    }
+
+    var disabled = true;
+    var updateButton = null;
+    if (loris.userHasPermission('candidate_parameter_edit')) {
+      disabled = false;
+      updateButton = React.createElement(ButtonElement, { label: "Update" });
+    }
+    var reasonDisabled = true;
+    var reasonRequired = false;
+    if (this.state.formData.flagged_caveatemptor === "true") {
+      reasonDisabled = false;
+      reasonRequired = true;
+    }
+
+    var reasonKey;
+    var specifyOther = null;
+    var otherDisabled = true;
+    var otherRequired = false;
+    for (var key in this.state.Data.caveatReasonOptions) {
+      if (this.state.Data.caveatReasonOptions.hasOwnProperty(key)) {
+        if (this.state.Data.caveatReasonOptions[key] === "Other") {
+          reasonKey = key;
+          break;
+        }
+      }
+    }
+
+    if (this.state.formData.flagged_reason === reasonKey) {
+      otherRequired = true;
+      otherDisabled = false;
+    }
+
+    if (this.state.formData.flagged_caveatemptor === "false") {
+      reasonDisabled = true;
+      reasonRequired = false;
+      otherDisabled = true;
+      otherRequired = false;
+      this.state.formData.flagged_reason = '';
+      this.state.formData.flagged_other = '';
```

Comment:
Also, to answer your question: 
1. You should use this.setState() to set the state
2. You can't set the state inside render as it results in 'infinite loop'